### PR TITLE
Bump to .NET 8 GA package refs

### PR DESCRIPTION
**Pull request by rainersigwald**
Update `dotnet/runtime` dependencies to the 8.0 versions.

- darc update-dependencies --id 199769
- Add Tasks.Dataflow and Channels deps
- Binding redirect updates to 8.0 versions

### CODE REVIEW

**Comment by rainersigwald** [comment]
Hey @dotnet/source-build-internal, what's the right thing to do here to fix up sourcebuild? This is for a component of 8.0.2xx that wants to reference "latest public" 8.0.0 packages (ideally without adding edges to the repo graph because we don't need to, SDK/installer construction will provide the latest for us regardless of what we reference in this repo and we can't push the .NET Framework side of our deps past whatever slightly-old version VS uses).

**Comment by mthalman** [comment]
> Hey @dotnet/source-build-internal, what's the right thing to do here to fix up sourcebuild? This is for a component of 8.0.2xx that wants to reference "latest public" 8.0.0 packages (ideally without adding edges to the repo graph because we don't need to, SDK/installer construction will provide the latest for us regardless of what we reference in this repo and we can't push the .NET Framework side of our deps past whatever slightly-old version VS uses).

The corresponding versions in https://github.com/dotnet/msbuild/blob/vs17.9/eng/SourceBuildPrebuiltBaseline.xml should be updated.

**Comment by rainersigwald** [comment]
@mthalman and extend the pattern for the new ones, rather than using the suggested one that sourcebuild spit out?

**Review by Forgind — APPROVED** [review]

**Review by maridematte — APPROVED** [review]

File: eng/SourceBuildPrebuiltBaseline.xml [hunk 1]
```diff
@@ -7,11 +7,15 @@
 
     <!-- Baseline 7.0 dependencies until msbuild targets net8 and uses a net8 arcade, SBRP, etc.
          These cannot be added to 7.0 SBRP, because they would are produced in the 7.0 build. -->
```
Comment (akoeplinger) [inline eng/SourceBuildPrebuiltBaseline.xml:9]: does the comment still apply?

Comment (rainersigwald) [reply eng/SourceBuildPrebuiltBaseline.xml:9]: No, I don't think so. I'll follow up in a later PR (since this is delaying VS work I want to get it in).

**Pull request merged**